### DISCUSSION TITLE
[IMPROVEMENT]Use CD/CI Variables for Deployer

### DISCUSSION THREAD

**Discussion by plexmaniax** [deployphp/deployer#3315] [General]
Am I right, there is no option to use CD/CI Variables in deployer, we always need a file like project.yaml where secrets are saved in plain text?

I'll tried it this way

host('staging')
    ->set('labels', ['stage' => 'staging'])
    ->set('hostname', getenv('SSH_HOST'))
    ->set('remote_user', getenv('SSH_USER'))
    ->set('http_user', getenv('SSH_USER'))
    ->set('deploy_path', getenv('DEPLOY_PATH'))
    ->set('port', (int) getenv('SSH_PORT'))
    ->set('bin/php', function() {
        return getenv('REMOTE_PHP_BINARY');
    })

But it seems a little bit unhandy. Is there a way to reuse CD/CI Variables not only as DOTENV ?

**Comment by antonmedv** [comment]
This is the way:

    getenv('SSH_PORT')
